cmd/autoconfigbrancher/main.go

```diff
@@ -155,66 +144,85 @@ func main() {
+		},
+	}
+
+	stdout := bumper.HideSecretsWriter{Delegate: os.Stdout, Censor: sa}
+	stderr := bumper.HideSecretsWriter{Delegate: os.Stderr, Censor: sa}
 	author := fmt.Sprintf("%s <%s>", o.gitName, o.gitEmail)
-	commitIfNeeded(fmt.Sprintf("determinize-ci-operator --current-release %s --future-release %s", o.currentRelease, strings.Join(o.futureReleases.Strings(), ",")), author)
-
-	cmd = "/usr/bin/config-brancher"
-	run(cmd, args...)
+	commitsCounter := 0
 
-	commitIfNeeded(fmt.Sprintf("config-brancher --current-release %s --future-release %s", o.currentRelease, strings.Join(o.futureReleases.Strings(), ",")), author)
-
-	cmd = "/usr/bin/ci-operator-prowgen"
-	args = []string{"--from-dir", "./ci-operator/config", "--to-dir", "./ci-operator/jobs"}
-	run(cmd, args...)
-
-	commitIfNeeded("ci-operator-prowgen --from-dir ./ci-operator/config --to-dir ./ci-operator/jobs", author)
-
-	cmd = "/usr/bin/ci-operator-config-mirror"
-	args = []string{"--config-path", "./ci-operator/config", "--to-org", "openshift-priv"}
-	run(cmd, args...)
-
-	commitIfNeeded("ci-operator-config-mirror --config-path ./ci-operator/config --to-org openshift-priv", author)
-
-	cmd = "/usr/bin/ci-operator-prowgen"
-	args = []string{"--from-dir", "./ci-operator/config", "--to-dir", "./ci-operator/jobs"}
-	run(cmd, args...)
-
-	commitIfNeeded("ci-operator-prowgen --from-dir ./ci-operator/config --to-dir ./ci-operator/jobs", author)
-
-	cmd = "/usr/bin/private-prow-configs-mirror"
-	args = []string{"--release-repo-path", "."}
-	run(cmd, args...)
-	commitIfNeeded("private-prow-configs-mirror --release-repo-path .", author)
-
-	cmd = "./hack/ensure_job_cluster.py"
-	args = []string{"-w=true"}
-	run(cmd, args...)
+	for i, step := range steps {
+		commited, err := runAndCommitIfNeeded(stdout, stderr, author, step.command, step.arguments)
+		if err != nil {
+			logrus.WithError(err).Fatalf("step[%d]: failed to run command and commit the changes", i)
+		}
 
-	cmd = "/usr/bin/sanitize-prow-jobs"
-	args = []string{"--prow-jobs-dir", "./ci-operator/jobs"}
-	run(cmd, args...)
-	commitIfNeeded("sanitize-prow-jobs  --prow-jobs-dir ./ci-operator/jobs", author)
+		if commited {
+			commitsCounter++
+		}
+	}
 
-	if count == 0 {
+	if commitsCounter == 0 {
 		logrus.Info("no new commits, existing ...")
-		return
+		os.Exit(0)
 	}
 
-	remoteBranch := "auto-config-brancher"
 	title := fmt.Sprintf("%s by auto-config-brancher job at %s", matchTitle, time.Now().Format(time.RFC1123))
-	cmd = "git"
-	args = []string{"push", "-f", fmt.Sprintf("https://%s:%s@github.com/%s/%s.git", o.githubLogin,
-		string(sa.GetTokenGenerator(o.GitHubOptions.TokenPath)()), o.githubLogin, githubRepo),
-		fmt.Sprintf("HEAD:%s", remoteBranch)}
-	run(cmd, args...)
+	if err := bumper.GitCommitAndPush(fmt.Sprintf("https://%s:%s@github.com/%s/%s.git", o.githubLogin, string(sa.GetTokenGenerator(o.GitHubOptions.TokenPath)()), o.githubLogin, githubRepo), remoteBranch, o.gitName, o.gitEmail, title, stdout, stderr); err != nil {
```

Comment:
We already committed, I don't think we should do it again. Will this error if there are no changes?

Reply:
We `commited` but we didn't push the changes in the fork's branch.

Reply:
@alvaroaleman you are right on this one. That will fail.... I going to fix this.